File: mobile/android/oxo-android-ben31/src/app/src/main/java/com/securebank/app/AccountActivity.java
package com.securebank.app;

import androidx.appcompat.app.AppCompatActivity;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class AccountActivity extends AppCompatActivity {

    private TextView accountNumberText;
    private TextView accountTypeText;
    private TextView balanceText;
    private TextView holderNameText;
    private Button backButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_account);

        accountNumberText = findViewById(R.id.accountNumberText);
        accountTypeText = findViewById(R.id.accountTypeText);
        balanceText = findViewById(R.id.balanceText);
        holderNameText = findViewById(R.id.holderNameText);
        backButton = findViewById(R.id.backButton);

        loadAccountInfo();

        backButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    private void loadAccountInfo() {
        SharedPreferences prefs = getSharedPreferences("bank_data", MODE_PRIVATE);
        String accountNumber = prefs.getString("account_number", "****-****-****-****");
        String balance = prefs.getString("current_balance", "0.00");
        String username = prefs.getString("logged_in_user", "Account Holder");

        accountNumberText.setText("Account Number: " + accountNumber);
        accountTypeText.setText("Account Type: Checking Account");
        balanceText.setText("Available Balance: $" + balance);
        holderNameText.setText("Account Holder: " + username.toUpperCase());
    }
}

File: mobile/android/oxo-android-ben69/src/app/src/main/java/co/ostorlab/cloudshare/AccountActivity.kt
package co.ostorlab.cloudshare

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class AccountActivity : AppCompatActivity() {
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_account)
        
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = "Account Settings"
        
        setupUI()
    }
    
    private fun setupUI() {
        // User info
        findViewById<TextView>(R.id.tv_user_name).text = "John Smith"
        findViewById<TextView>(R.id.tv_user_email).text = "john.smith@company.com"
        findViewById<TextView>(R.id.tv_plan_type).text = "CloudShare Pro - Business"
        findViewById<TextView>(R.id.tv_member_since).text = "Member since March 2023"
        
        // Buttons
        findViewById<Button>(R.id.btn_edit_profile).setOnClickListener {
            // Edit profile functionality
        }
        
        findViewById<Button>(R.id.btn_change_password).setOnClickListener {
            // Change password functionality
        }
        
        findViewById<Button>(R.id.btn_security_settings).setOnClickListener {
            startActivity(Intent(this, SecurityActivity::class.java))
        }
        
        findViewById<Button>(R.id.btn_logout).setOnClickListener {
            // Logout functionality
            finish()
        }
    }
    
    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
